app/models/document.rb

```diff
@@ -53,9 +53,30 @@ class Document < ApplicationRecord
 
   before_save :generate_pdf_tmpfile
   before_save :add_converted_preview_document
+  before_save :create_checksum
 
   validate :documents_count
 
+  # TODO: Remove after moving to Active Storage
```

Comment:
Appreciate that this is an opimization for something that is temporary (famous last words) but we could create a helper/mixin like
```
module CheckSummable
    def add_checksum(checksum_field, io)
      Digest::MD5.new.tap do |checksum|
        while (chunk = io.read(5.megabytes))
          checksum << chunk
        end
        
        send("#{checksum_field}=", checksum)

        io.rewind
      end.base64digest
     rescue Errno::ENOENT
       send("#{checksum_field}=", nil)
    end
end
```

and eg. for use. For Document you could just loop over the the two documents i guess

```
class Message
  ....
  before_save :populate_checksum

  def polulate_checksum
    return if attachment_file_name.nil?
    io = Paperclip.io_adapters.for(attachment)

     add_checksum(:as_attachment_checksum, io)
  end

```

Reply:
do we also need to handle long name error
```
rescue Errno::ENAMETOOLONG
    log('FilenameTooLong')....blah
    send("#{checksum_field}=", nil)
 ```

Reply:
I have moved this into a separate module now and have it logging (and not setting the checksum) when there are errors.